tests/Aspire.Pomelo.EntityFrameworkCore.MySql.Tests/AspireEFMySqlExtensionsTests.cs

```diff
@@ -109,11 +109,51 @@ public void CanConfigureDbContextOptions()
         var actualConnectionString = context.Database.GetDbConnection().ConnectionString;
         Assert.Equal(ConnectionString + ";Allow User Variables=True;Default Command Timeout=123;Use Affected Rows=False", actualConnectionString);
 
-        // ensure the max retry count from config was respected
+        // ensure the retry strategy is enabled and set to its default value
         Assert.NotNull(extension.ExecutionStrategyFactory);
         var executionStrategy = extension.ExecutionStrategyFactory(new ExecutionStrategyDependencies(new CurrentDbContext(context), context.Options, null!));
         var retryStrategy = Assert.IsType<MySqlRetryingExecutionStrategy>(executionStrategy);
-        Assert.Equal(304, retryStrategy.MaxRetryCount);
+        Assert.Equal(new WorkaroundToReadProtectedField(context).MaxRetryCount, retryStrategy.MaxRetryCount);
+
+#pragma warning restore EF1001 // Internal EF Core API usage.
+    }
+
+    [Fact]
+    public void CanConfigureDbContextOptionsWithoutRetry()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>("Aspire:Pomelo:EntityFrameworkCore:MySql:ServerVersion", "8.2.0-mysql"),
+            new KeyValuePair<string, string?>("ConnectionStrings:mysql", ConnectionString),
+            new KeyValuePair<string, string?>("Aspire:Pomelo:EntityFrameworkCore:MySql:Retry", "false")
+        ]);
+
+        builder.AddMySqlDbContext<TestDbContext>("mysql", configureDbContextOptions: optionsBuilder =>
+        {
+            optionsBuilder.UseMySql(new MySqlServerVersion(new Version(8, 2, 0)), mySqlBuilder =>
+            {
+                mySqlBuilder.CommandTimeout(123);
+            });
+        });
+
+        var host = builder.Build();
+        var context = host.Services.GetRequiredService<TestDbContext>();
+
+#pragma warning disable EF1001 // Internal EF Core API usage.
+
+        var extension = context.Options.FindExtension<MySqlOptionsExtension>();
+        Assert.NotNull(extension);
+
+        // ensure the command timeout was respected
+        Assert.Equal(123, extension.CommandTimeout);
+
+        // Other EF components check the connection string is respected but MySqlConnector
+        // is adding default properties to it.
+        //var actualConnectionString = context.Database.GetDbConnection().ConnectionString;
+        //Assert.Equal(ConnectionString, actualConnectionString);
```

Comment:
Can we do what the other tests are doing?

https://github.com/dotnet/aspire/blob/773f4ad74376946ebdbfee2d3d7171056cd7d722/tests/Aspire.Pomelo.EntityFrameworkCore.MySql.Tests/AspireEFMySqlExtensionsTests.cs#L53